src/components/ProductList.jsx

```diff
@@ -4,33 +4,43 @@ import Pagination from "./Pagination";
 import "./ProductList.css";
 import SearchInput from "./SearchInput";
 import { useNavigate } from "react-router-dom";
+import { getProducts } from "../api/productApi";
+import Dropdown from "./Dropdown";
 
 /**
  *  전체 상품 목록을 카드 형태로 표시하는 컴포넌트 입니다.
  *  검색어 기능, 정렬, 페이징 기능이 포함되어있습니다.
  */
-const ProductList = ({ products }) => {
+const ProductList = () => {
+  const [products, setProducts] = useState([]);
   const [orderBy, setOrderBy] = useState("recent");
   const [page, setPage] = useState(1);
   const [searchKeyword, setSearchKeyword] = useState("");
+  const [isMobile, setIsMobile] = useState(false);
 
   const nav = useNavigate(); // 페이지 이동을 위한 hook 사용
 
-  const pageSize = 10; // 한 페이지에 보여줄 상품 수
-
   const [responsivePageSize, setResponsivePageSize] = useState(10); // 기본값 10
 
-  // option 값에 따라 정렬된 상품 리스트를 보여줌 (desc)
-  const sortedProducts = [...products].sort((a, b) => {
-    if (orderBy === "favorite") return b.favoriteCount - a.favoriteCount;
-    return new Date(b.createdAt) - new Date(a.createdAt);
-  });
+  useEffect(() => {
+    const params = {
+      page: 1,
+      pageSize: 300,
+      orderBy,
+    };
+    if (searchKeyword) {
+      params.keyword = searchKeyword;
+    }
```

Comment:
💬 여담
아래 코드도 동일동작합니다~
```suggestion
    const params = {
      page: 1,
      pageSize: 300,
      orderBy,
      ...(searchKeyword && { keyword: searchKeyword })
    };
```